cpu/esp8266/Makefile.include

```diff
@@ -118,48 +119,78 @@ ifeq ($(FLASH_MODE), qout)
+LINKFLAGS += -T$(RIOTCPU)/$(CPU)/ld/esp8266.riot-os.ld
+LINKFLAGS += -T$(RIOTCPU)/$(CPU)/ld/esp8266.peripherals.ld
+
+ifneq (, $(filter esp_idf_heap, $(USEMODULE)))
+    LINKFLAGS += -Wl,-wrap=_malloc_r
+    LINKFLAGS += -Wl,-wrap=_calloc_r
+    LINKFLAGS += -Wl,-wrap=_free_r
+    LINKFLAGS += -Wl,-wrap=_realloc_r
+endif
 
 # The ELFFILE is the base one used for flashing
 FLASHFILE ?= $(ELFFILE)
 
 # configure preflasher to convert .elf to .bin before flashing
-FLASH_SIZE = -fs 8m
-PREFLASHER ?= esptool.py
-PREFFLAGS  ?= elf2image $(FLASH_SIZE) $(FLASHFILE)
-FLASHDEPS  ?= preflash
-
-# flasher configuration
-ifeq ($(QEMU), 1)
-    FLASHER = cat
-    FFLAGS += $(FLASHFILE)-0x00000.bin /dev/zero | head -c $$((0x10000)) | cat -
-    FFLAGS += $(FLASHFILE)-0x10000.bin /dev/zero | head -c $$((0xfc000)) | cat -
-    FFLAGS += $(RIOTCPU)/$(CPU)/bin/esp_init_data_default.bin > $(FLASHFILE).bin
+FLASH_MODE = dout  # FIX configuration, DO NOT CHANGE
+FLASH_FREQ = 26m   # FIX configuration, DO NOT CHANGE
+FLASH_SIZE ?= 1MB
+
+PREFLASHER ?= $(ESP8266_SDK_DIR)/components/esptool_py/esptool/esptool.py
+PREFFLAGS  = --chip esp8266 elf2image
+PREFFLAGS += --flash_mode $(FLASH_MODE) --flash_size $(FLASH_SIZE)
+PREFFLAGS += --flash_freq $(FLASH_FREQ) --version 3
+PREFFLAGS += -o $(FLASHFILE).bin $(FLASHFILE);
+PREFFLAGS += echo "" > $(BINDIR)/partitions.csv;
+PREFFLAGS += echo "nvs, data, nvs, 0x9000, 0x6000" >> $(BINDIR)/partitions.csv;
+PREFFLAGS += echo "phy_init, data, phy, 0xf000, 0x1000" >> $(BINDIR)/partitions.csv;
+PREFFLAGS += echo -n "factory, app, factory, 0x10000, " >> $(BINDIR)/partitions.csv;
+PREFFLAGS += ls -l $(FLASHFILE).bin | awk '{ print $$5 }' >> $(BINDIR)/partitions.csv;
+
+PREFFLAGS += python $(RIOTCPU)/$(CPU)/vendor/esp-idf/partition_table/gen_esp32part.py
+PREFFLAGS += --verify $(BINDIR)/partitions.csv $(BINDIR)/partitions.bin
+FLASHDEPS  = preflash
+
+ifneq (, $(filter esp_log_colored, $(USEMODULE)))
+    BOOTLOADER ?= bootloader_dout_115200_color.bin
+else
+    BOOTLOADER ?= bootloader_dout_115200_no_color.bin
+endif
+
+ifneq (, $(filter esp_qemu, $(USEMODULE)))
+    FLASHER = dd
+    FFLAGS += if=/dev/zero bs=1M count=1 | tr "\\000" "\\377" > tmp.bin &&
+    FFLAGS += cat $(RIOTCPU)/$(CPU)/bin/$(BOOTLOADER) tmp.bin |
+    FFLAGS += head -c $$((0x8000)) |
+    FFLAGS += cat - $(BINDIR)/partitions.bin tmp.bin |
+    FFLAGS += head -c $$((0x10000)) |
+    FFLAGS += cat - $(FLASHFILE).bin tmp.bin |
+    FFLAGS += head -c $$((0xfc000)) |
+    FFLAGS += cat - $(RIOTCPU)/$(CPU)/bin/esp_init_data_default.bin tmp.bin |
+    FFLAGS += head -c $$((0x100000)) > $(BINDIR)/esp8266flash.bin && rm tmp.bin
 else
-    FLASH_MODE ?= dout
     export PROGRAMMER_SPEED ?= 460800
     FLASHER = esptool.py
-    FFLAGS += -p $(PORT) -b $(PROGRAMMER_SPEED) write_flash
-    FFLAGS += -fm $(FLASH_MODE)
-    FFLAGS += 0 $(FLASHFILE)-0x00000.bin
-    FFLAGS += 0x10000 $(FLASHFILE)-0x10000.bin; esptool.py -p $(PORT) run
```

Comment:
This is great, it removes the two invocations of `esptool.py`. On some local setup, the `$(PORT)` disappeared after the first command and was not there before the call.